spec/integration/outputs/index_version_spec.rb

```diff
@@ -0,0 +1,88 @@
+require_relative "../../../spec/es_spec_helper"
+require "logstash/outputs/elasticsearch"
+
+
+describe "Versioned indexing", :integration => true, :version_greater_than_equal_to_2x => true do
+  require "logstash/outputs/elasticsearch"
+
+  def get_es_output( options={} )
+    settings = {
+      "manage_template" => true,
+      "index" => "logstash-index",
+      "template_overwrite" => true,
+      "hosts" => get_host_port(),
+      "action" => "index",
+      "script_lang" => "groovy"
+    }
+    LogStash::Outputs::ElasticSearch.new(settings.merge!(options))
+  end
+
+  before :each do
+    @es = get_client
+    # Delete all templates first.
+    # Clean ES of data before we start.
+    @es.indices.delete_template(:name => "*")
+    # This can fail if there are no indexes, ignore failure.
+    @es.indices.delete(:index => "*") rescue nil
+    @es.indices.refresh
+  end
+
+  context "when index only" do
+    it "should default to ES version" do
+      subject = get_es_output({ 'document_id' => "123" })
+      subject.register
```

Comment:
Love that you wrote tests here! Same deal here WRT `before` blocks and better leveraging `let`.